Removed comma separator

paco-arana opened the pull request:
**Description**

Fixes #5260

area: `fr.free.nrw.commons.description.DescriptionEditActivity#updateDescription`

The comma separator between descriptions in different languages has been removed from `fr.free.nrw.commons.description.DescriptionEditActivity#updateDescription` line 154. This also made line 157 redundant as its function was deleting the very last comma after all the descriptions were formatted.

**Tests performed**

Tested beta build on Xiaomi Redmi 9 using Android 11.

nicolas-raoul reviewed (changes requested):
I uploaded a file using your branch and still see a comma, that's strange: https://commons.wikimedia.org/wiki/File:Binh_Tay_market_2.jpg

Maybe DescriptionEditActivity is only used when modifying an existing file, rather than when uploading a new file?

Thanks! :-)

nicolas-raoul commented on app/src/main/java/fr/free/nrw/commons/contributions/Contribution.kt:
```diff
@@ -112,5 +112,6 @@ data class Contribution constructor(
         fun formatDescriptions(descriptions: List<UploadMediaDetail>) =
             descriptions.filter { it.descriptionText.isNotEmpty() }
                 .joinToString { "{{${it.languageCode}|1=${it.descriptionText}}}" }
+                .replace("}}, ", "}}")
```
Would not it be better to not add the comma in the first place?

paco-arana replied:
I tried that first but I was having a hard time determining where the comma was actually originating from, I'll dig a little more until I can find it. 

Thanks for the quick response, I apologize if I'm making you work more than needed.

paco-arana replied:
Okay, I think this one should do it, I should've checked the default separator for the .joinToString first before trying to look for the comma everywhere else.

nicolas-raoul reviewed (commented):
Looks great! I will test your pull request as soon as possible. 🙂

nicolas-raoul reviewed (changes requested):
Uploading now works great! :-)
Unfortunately editing adds the commas:
https://commons.wikimedia.org/w/index.php?title=File%3AKaba_amphibious_bus_in_Odaiba.jpg&diff=784750015&oldid=784749122
Would you mind checking?

paco-arana commented:
I seem to have caused that myself when I modified DescriptionEditActivity#updateDescription, I added the deleted line back in and now it's working as it should:

https://commons.wikimedia.org/w/index.php?title=File:Catedral_de_Aguascalientes.jpg&diff=prev&oldid=784872825

nicolas-raoul reviewed (approved):
Now working great for both upload and modification.
Example: https://commons.wikimedia.org/wiki/File:Central_Building,_Sakura_Tower.jpg
Thanks a lot!
